designs/2022-forbid-disable/README.md

```diff
@@ -0,0 +1,125 @@
+
+This error will **not** have a quick fix available
+
+
+## Documentation
+
+We will want to update the documentation to reflect the new changes. The announcement can be integrated into the eslintrc update announcement.
+
+## Drawbacks
+
+this adds complexity to the already complex eslintrc experience. Users could be confused by the interaction with `forbid` and `noInlineConfig`.
+
+This will have a very minor impact on perf.
+
+## Backwards Compatibility Analysis
+
+This should integrate into the planned eslintrc changes and therefore will not be backwards compatible. However, this is acceptable since that will be a breaking change that is already approved and planned.
+
+## Alternatives
+
+instead of adding a state to `error/warn` we could instead add another optional field to the rule config. This would enable a rule to be `warn` and have directives disabled. However, I believe that this actually adds more bloat to eslintrc which is why I chose the above design
+
+There are plugins such as [eslint-comments/no-restricted-disable](https://mysticatea.github.io/eslint-plugin-eslint-comments/rules/no-restricted-disable.html) that implement this concept. However, since these are just other rules, one can easily disable them. This results in scenarios such as
+
+```
+// eslint-disable-next-line eslint-comments/no-restricted-disable
+// eslint-disable no-undef
+```
+In large PRs, reviewers often miss these comments. In a large repo that I work on these scenario has happened over 30 times
+
+Another alternative that requires 0 code is to encourage users to set up a second eslintrc and run lint twice. Once normally with regular rules, and once with `no-inline-config` with just the rules that are restricted. However, **this will not integrate into vscode error reporting**. Which will result in users pushing code to PR phase only to fail a gate that checks for restricted disables. This solution also results in a lot of bloat and configuration overhead for anyone who wants this functionality.
+
+## Open Questions
+
+<!--
+    This section is optional, but is suggested for a first draft.
+
+    What parts of this proposal are you unclear about? What do you
+    need to know before you can finalize this RFC?
+
+    List the questions that you'd like reviewers to focus on. When
+    you've received the answers and updated the design to reflect them, 
+    you can remove this section.
+-->
+
+- How can I gracefully integrate into the eslintrc updates?
+- How can I test runtime to ensure that this doesn't affect perf?
+- What is the testing strategy that ESLint community prefers?
+
+## Help Needed
+
+<!--
+    This section is optional.
+
+    Are you able to implement this RFC on your own? If not, what kind
+    of help would you need from the team?
+-->
+- I can implement this independently
+- Will the RFC still need to be open for 21 days if the main contributors all chime in?
```

Comment:
Yes. Initial commenting is always at least 21 days to allow for community feedback.